### DISCUSSION TITLE
Make use of this in C

### DISCUSSION THREAD

**Discussion by ahiser1117** [animetosho/md5-optimisation#2] [Q&A]
Hello Authors,

I was hoping to implement this into a C program I've been writing. My code is in C so I need to somehow translate the .h files so I can call them. I made the first go at it but I don't know enough about MD5 or inline assembly to do it. The goal is to simply input a message and receive a hash as fast as possible. Do you have suggestions about how I could do this?

Thanks

**Comment by animetosho** [comment]
Thanks for your interest!

This project is intended more to demonstrate to developers, implementing MD5, what can be done to improve their implementation, as opposed to an easy-to-use library for those who need an MD5 function in their application.  
If you're looking for the latter, I'd suggest looking elsewhere for something more aimed at being used. (OpenSSL is mentioned in the article, which should work, but if you're only after MD5, it may be a bit overwhelming)

If you do really want to use this implementation though, you can copy bits from md5.cpp, and invoke `md5` [like this](https://github.com/animetosho/md5-optimisation/blob/31fc4e985ca9efbed5810c665d415b19e089ebee/md5.cpp#L68). The code is C++, but hopefully shouldn't be too difficult to translate to C. Also note that there are a number of MD5 implementations here (as it's testing the effect of various optimisations), so you'll need to pick one that you want to use.
